jose/generate.go

```diff
@@ -38,28 +41,41 @@ func GenerateJWKFromPEM(filename string) (*JSONWebKey, error) {
 	}
 
 	switch key := key.(type) {
-	case *rsa.PrivateKey, *rsa.PublicKey:
-		return &JSONWebKey{Key: key}, nil
-	case *ecdsa.PrivateKey:
-		return &JSONWebKey{
-			Key:       key,
-			Algorithm: getECAlgorithm(key.Curve),
-		}, nil
-	case *ecdsa.PublicKey:
+	case *rsa.PrivateKey, *rsa.PublicKey, *ecdsa.PrivateKey, *ecdsa.PublicKey, ed25519.PrivateKey, ed25519.PublicKey:
```

Comment:
I would separate RSA keys from the others so it's easier to see that the algorithm is not automatically set in those cases.